docs/blog/2019-03-25-behind-the-scenes-what-makes-gatsby-great/index.md

```diff
@@ -0,0 +1,496 @@
+
+But, but... you exclaim! What about progressive enhancement? What about mobile? What about devices that are data constrained? You better believe we've got 'em all handled.
+
+**Progressive Enhancement**
+
+The feature seamlessly falls back to default behavior if `IntersectionObserver` is not detected. In browsers that support it ([most!](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Browser_compatibility)) we'll send the idle `prefetch`. In browsers that don't, we'll only use the hover interaction. As browsers continue to implement more of these modern APIs, you can rest assured that your Gatsby application will be using the best and most useful, in a way that degrades seamlessly for your end users.
+
+**Mobile**
+
+Most mobile browsers support `IntersectionObserver` but none (of course!) support the `onMouseEnter` event. This means that the strong, non-idle fetch will not be triggered on mobile. This is not necessarily ideal--but the beauty of the feature is that the `prefetch` generally takes care of the necessary resource(s) being avalable.
+
+**Data-constrained devices**
+
+One of the most appealing things about using a framework like Gatsby is that we can bake-in micro-optimizations and make these available in incremental updates to all of _our_ users making the performance optimizations available to all of _your_ users. In particular, later versions of Chrome support an API that grants network information like type of connection (e.g. `slow-2g`, `2g`, etc.), whether the browser has a `saveData` hint enabled, and more. With the help of [@addyosmani](https://twitter.com/addyosmani) we've baked this into our preloading logic. On devices with these hints or on slow connections, we disable the preloading to save data! It looks a little something like:
+
+```js
+// Skip prefetching if we know user is on slow or constrained connection
+if (`connection` in navigator) {
+  if ((navigator.connection.effectiveType || ``).includes(`2g`)) {
+    return false
+  }
+  if (navigator.connection.saveData) {
+    return false
+  }
+}
+```
+
+See the [code implementing this functionality here](https://github.com/gatsbyjs/gatsby/blob/be023447eb0f5bd4612a447098fc85c7afdf0cdf/packages/gatsby/cache-dir/loader.js#L227-L235).
+
+### `srcset` powering Responsive Images
+
+Whew. It's been quite a ride. Finally, I'd like to discuss one of my favorite Gatsby components, specifically [`gatsby-image`](/packages/gatsby-image/). This component _also_ registers an `IntersectionObserver`, and uses it in an interesting, performance-focused manner. Let's get to it.
+
+#### `gatsby-image`
+
+There are far, far, far too many otherwise decently performing websites that load a gargantuan 5Mb image request as a hero image. But--thankfully, this isn't you! You use `gatsby-image` and you get all of the following:
```

Comment:
```suggestion
There are far, far, far too many otherwise decently performing websites that load a gargantuan 5Mb image request as a hero image. But--thankfully, this isn't you if you're using `gatsby-image`. With it you get all of the following:
```